lib/ruby_lsp/requests/goto_relevant_file.rb

```diff
@@ -0,0 +1,83 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Requests
+    # Goto Relevant File is a custom [LSP
+    # request](https://microsoft.github.io/language-server-protocol/specification#requestMessage)
+    # that navigates to the relevant file for the current document.
+    # Currently, it supports source code file <> test file navigation.
+    class GotoRelevantFile < Request
+      extend T::Sig
+
+      TEST_KEYWORDS = ["test", "spec", "integration_test"]
+
+      sig { params(path: String).void }
+      def initialize(path)
+        super()
+
+        @workspace_path = T.let(Dir.pwd, String)
+        @path = T.let(path.delete_prefix(@workspace_path), String)
+      end
+
+      sig { override.returns(T::Array[String]) }
+      def perform
+        find_relevant_paths
+      end
+
+      private
+
+      sig { returns(T::Array[String]) }
+      def find_relevant_paths
+        workspace_path = Dir.pwd
```

Comment:
Let's take in `GlobalState` and use its `workspace_path` method to get this value. I think `Requests::Rename` is a good reference.